extensions/cloudfiles-extensions/src/main/java/io/druid/storage/cloudfiles/CloudFilesByteSource.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.storage.cloudfiles;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.jclouds.io.Payload;
+
+import com.google.common.base.Throwables;
+import com.google.common.io.ByteSource;
+
+public class CloudFilesByteSource extends ByteSource {
+
+	final private CloudFilesObjectApiProxy objectApi;
+	final private String path;
+	private Payload payload;
+
+	public CloudFilesByteSource(CloudFilesObjectApiProxy objectApi, String path) {
+		this.objectApi = objectApi;
+		this.path = path;
+		this.payload = null;
+	}
+
+	public void closeStream() throws IOException {
+		if (payload != null) {
+			payload.close();
+		}
+	}
+
+	@Override
+	public InputStream openStream() throws IOException {
+		payload = objectApi.get(path).getPayload();
```

Comment:
Do you say this in order to avoid the calling of `get` again if it is already present? It seems a good idea, it just requires that I reset `payload` to `null` when I close it, right?